### DISCUSSION TITLE
How to use javascript function from external file in twig in Symfony 6

### DISCUSSION THREAD

**Discussion by ScorpionZZZ** [symfony/symfony#49045] [Q&A]
I try to use function in external js file in twig template... but isn't work (

/assets/app_departments.js
```
export function test(m) { console.log(m); }
```

webpack.config.js
```
....
.addEntry('app_departments', './assets/app_departments.js')
...
```

test.twig.html
```
{% extends 'base.html.twig' %}

{% block javascripts %}
    {{ parent() }}
    {{ encore_entry_script_tags('app_departments') }}
{% endblock %}

{% block body %}
    <script>
        test('passed');
    </script>
{% endblock %}
```

Browser console:
> ReferenceError: Can't find variable: test

**Comment by MatTheCat** [comment]
It may be the script tag is added with the `defer` attribute, so it is not executed before the DOM is loaded, so you cannot use it before.

You could remove the attribute but it would probably be better to implement your logic in a separate JavaScript file rather than your Twig template.
